deeppavlov/core/agent/dialog_logger.py

```diff
@@ -0,0 +1,134 @@
+# Copyright 2017 Neural Networks and Deep Learning lab, MIPT
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+from pathlib import Path
+from datetime import datetime
+from typing import Any, Optional, Hashable
+
+from deeppavlov.core.common.log import get_logger
+from deeppavlov.core.common.paths import get_configs_path
+from deeppavlov.core.common.file import read_json
+from deeppavlov.core.data.utils import jsonify_data
+from deeppavlov.core.agent.rich_content import RichMessage
+
+LOGGER_CONFIG_FILENAME = 'dialog_logger_config.json'
+LOG_TIMESTAMP_FORMAT = '%Y-%m-%d_%H-%M-%S_%f'
+
+log = get_logger(__name__)
+
+
+class DialogLogger:
+    """DeepPavlov dialog logging facility.
+
+    DialogLogger is an entity which provides tools for dialogs logging.
+
+    Args:
+        enabled: DialogLogger on/off flag.
+        agent_name: Agent name which is used for organising log files.
+
+    Attributes:
+        agent_name: Agent name which is used for organising log files.
+        log_max_size: Maximum size of log file, kb.
+        self.log_file: Current log file object.
+    """
+    def __init__(self, enabled: bool=False, agent_name: Optional[str]=None):
+        self.config: dict = read_json(get_configs_path() / LOGGER_CONFIG_FILENAME)
+        self.enabled: bool = enabled or self.config['enabled']
+
+        if self.enabled:
+            self.agent_name: str = agent_name or self.config['agent_name']
+            self.log_max_size: int = self.config['logfile_max_size_kb']
+            self.log_file = self._get_log_file()
+            self.log_file.writelines('Agent initiated\n')
+
+    @staticmethod
+    def _get_timestamp_utc_str() -> str:
+        """Returns str converted current UTC timestamp.
+
+        Returns:
+            utc_timestamp_str: str converted current UTC timestamp.
+        """
+        utc_timestamp_str = datetime.strftime(datetime.utcnow(), LOG_TIMESTAMP_FORMAT)
+        return utc_timestamp_str
+
+    def _get_log_file(self):
+        """Returns opened file object for writing dialog logs.
+
+        Returns:
+            log_file: opened Python file object.
+        """
+        log_dir: Path = Path(self.config['log_path']).expanduser().resolve() / self.agent_name
+
+        dot_dp_path = Path('~/.deeppavlov').expanduser().resolve()
+        if dot_dp_path in log_dir.parents and dot_dp_path.is_file():
+            dot_dp_path.unlink()
```

Comment:
This logic should be moved somewhere higher, maybe `deeppavlov/__init__.py`?
We'll be downloading models and embeddings in `~/.deeppavlov` soon enough, so it's not only logging's problem

Reply:
Moved to `deeppavlov/__init__.py`